frontend/src/components/commons/TabNavigation.jsx

```diff
@@ -67,6 +68,25 @@ class TabNavigation extends Component {
     return isTabIdDisabled(this.props.disabledTabsArray, tabId);
   };
 
+  /* this function is called as soon as you start the test
+  it is used to update the states (currentTab and currentBody) with the given prop values */
+  componentWillReceiveProps(nextProps) {
```

Comment:
I don't think you actually need `firstRun` here. This works too and is much simpler.

```
  /* this function is called as soon as you start the test
  it is used to update the states (currentTab and currentBody) with the given prop values */
  componentWillReceiveProps(nextProps) {
    if (nextProps.currentTab !== this.state.currentTab) {
      this.setState({
        currentTab: nextProps.currentTab,
        currentBody: nextProps.tabSpecs[nextProps.currentTab].body
      });
    }
  }
```

Reply:
Yes we need `firstRun` state, otherwise when you select **Quit test** or **Submit test**, it changes the current tab to _Background_ tab.

![image](https://user-images.githubusercontent.com/23021242/56970830-05218680-6b36-11e9-8085-07c19ce7c408.png)

![image](https://user-images.githubusercontent.com/23021242/56970857-110d4880-6b36-11e9-8e0b-99be8030cc15.png)

Reply:
Alright I updated the naming to make this a little clearer, but I think there is an underlying bug in this that we'll need to find at some point if we make any other changes to the tabs.